cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -926,68 +992,6 @@ ExtensionIdRegistry::SubstraitCallToArrow DecodeConcatMapping() {
   };
 }
 
-ExtensionIdRegistry::SubstraitAggregateToArrow DecodeBasicAggregate(
```

Comment:
TODO: Revert this change after https://github.com/apache/arrow/pull/35608 is merged

Reply:
Done